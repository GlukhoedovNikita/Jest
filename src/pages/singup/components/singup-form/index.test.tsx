import renderWithRedux from '@common/utils/tests/renderWithRedux'
import { fireEvent, waitFor } from '@testing-library/dom'

import SingUpForm from './index'

describe('Sing up form', () => {
    it('render', () => {
        const component = renderWithRedux(<SingUpForm />)

        expect(component.getByTestId('singup-form')).toBeInTheDocument()
        expect(component.getByTestId('title')).toBeInTheDocument()
        expect(component.getByTestId('name')).toBeInTheDocument()
        expect(component.getByTestId('email')).toBeInTheDocument()
        expect(component.getByTestId('password')).toBeInTheDocument()
        expect(component.getByTestId('repeatPassword')).toBeInTheDocument()
        expect(component.getByTestId('button')).toBeInTheDocument()
    })

    it('snapshot', () => {
        const component = renderWithRedux(<SingUpForm />)

        expect(component).toMatchSnapshot()
    })

    it('change input', () => {
        const component = renderWithRedux(<SingUpForm />)
        const name = component.getByTestId('name')
        const email = component.getByTestId('email')
        const password = component.getByTestId('password')
        const repeatPassword = component.getByTestId('repeatPassword')

        const target = { target: { value: '1' } }
        fireEvent.change(name, target)
        fireEvent.change(email, target)
        fireEvent.change(password, target)
        fireEvent.change(repeatPassword, target)

        expect(name).toHaveValue('1')
        expect(email).toHaveValue('1')
        expect(password).toHaveValue('1')
        expect(repeatPassword).toHaveValue('1')
    })

    it('incorrect required', async () => {
        const component = renderWithRedux(<SingUpForm />)
        const form = component.getByTestId('singup-form')

        fireEvent.submit(form)

        expect(await component.findAllByText(/this field is required/i)).toHaveLength(4)
    })

    it('incorrect email', async () => {
        const component = renderWithRedux(<SingUpForm />)
        const form = component.getByTestId('singup-form')
        const email = component.getByTestId('email')

        const target = { target: { value: '123' } }
        fireEvent.change(email, target)
        fireEvent.submit(form)

        expect(await component.findAllByText(/incorrect mail/i)).toHaveLength(1)
    })

    it('password dont match', async () => {
        const component = renderWithRedux(<SingUpForm />)
        const form = component.getByTestId('singup-form')
        const password = component.getByTestId('password')
        const repeatPassword = component.getByTestId('repeatPassword')

        const tagretPassword = { target: { value: '34534' } }
        const tagretRepeatPassword = { target: { value: '345343' } }

        fireEvent.change(password, tagretPassword)
        fireEvent.change(repeatPassword, tagretRepeatPassword)
        fireEvent.submit(form)

        expect(await component.findByText(/passwords do not match/i)).toBeInTheDocument()
    })

    it('password min 3, max 15', async () => {
        const component = renderWithRedux(<SingUpForm />)
        const form = component.getByTestId('singup-form')
        const password = component.getByTestId('password')

        const targetMin = { target: { value: '4' } }

        fireEvent.change(password, targetMin)
        fireEvent.submit(form)

        expect(await component.findByText(/minimum field length - 3/i)).toBeInTheDocument()

        const targetMax = { target: { value: '1111122222333334' } }

        fireEvent.change(password, targetMax)
        fireEvent.submit(form)

        expect(await component.findByText(/maximum field length - 15/i)).toBeInTheDocument()
    })

    it('correct all input', async () => {
        const component = renderWithRedux(<SingUpForm />)
        const form = component.getByTestId('singup-form')
        const name = component.getByTestId('name')
        const email = component.getByTestId('email')
        const password = component.getByTestId('password')
        const repeatPassword = component.getByTestId('repeatPassword')

        const targetName = { tagret: { value: 'name' } }
        const targetEmail = { tagret: { value: 'test@gmail.ru' } }
        const targetPassword = { tagret: { value: '1234' } }
        const targetRepeatPassword = { tagret: { value: '1234' } }

        fireEvent.change(name, targetName)
        fireEvent.change(email, targetEmail)
        fireEvent.change(password, targetPassword)
        fireEvent.change(repeatPassword, targetRepeatPassword)
        fireEvent.submit(form)

        await waitFor(() => {
            expect(component.queryByText(/this field is required/i)).toBeNull()
            expect(component.queryByText(/incorrect mail/i)).toBeNull()
            expect(component.queryByText(/passwords do not match/i)).toBeNull()
            expect(component.queryByText(/minumum field length - 3/i)).toBeNull()
            expect(component.queryByText(/maximum field length - 15/i)).toBeNull()
        })
    })
})
