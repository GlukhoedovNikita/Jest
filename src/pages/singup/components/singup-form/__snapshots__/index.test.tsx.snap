// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sing up form snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        data-testid="singup-form"
      >
        <h2
          data-testid="title"
        >
          Sing Up
        </h2>
        <label>
          Name
          <input
            data-testid="name"
            name="name"
          />
        </label>
        <label>
          Email
          <input
            data-testid="email"
            name="email"
          />
        </label>
        <label>
          Password
          <input
            data-testid="password"
            name="password"
          />
        </label>
        <label>
          Repeat password
          <input
            data-testid="repeatPassword"
            name="repeatPassword"
          />
        </label>
        <button
          data-testid="button"
          type="submit"
        >
          Sing Up
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      data-testid="singup-form"
    >
      <h2
        data-testid="title"
      >
        Sing Up
      </h2>
      <label>
        Name
        <input
          data-testid="name"
          name="name"
        />
      </label>
      <label>
        Email
        <input
          data-testid="email"
          name="email"
        />
      </label>
      <label>
        Password
        <input
          data-testid="password"
          name="password"
        />
      </label>
      <label>
        Repeat password
        <input
          data-testid="repeatPassword"
          name="repeatPassword"
        />
      </label>
      <button
        data-testid="button"
        type="submit"
      >
        Sing Up
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
